homeassistant/config.py

```diff
@@ -1350,16 +1335,12 @@ def async_handle_component_errors(
     else:
         translation_key = ConfigErrorTranslationKey.INTEGRATION_CONFIG_ERROR
         errors = str(len(config_exception_info))
-        log_message = (
-            f"Failed to process component config for integration {domain} "
-            f"due to multiple errors ({errors}), check the logs for more information."
-        )
         placeholders = {
             "domain": domain,
             "errors": errors,
         }
     raise ConfigValidationError(
-        str(log_message),
+        translation_key,
```

Comment:
Shouldn't this happen in the constructor instead, like this?

```
    def __init__(
        self,
        exceptions: list[Exception],
        translation_domain: str | None = None,
        translation_key: str | None = None,
        translation_placeholders: dict[str, str] | None = None,
    ) -> None:
        """Initialize exception."""
        super().__init__(
            *(translation_key, exceptions),
            translation_domain=translation_domain,
            translation_key=translation_key,
            translation_placeholders=translation_placeholders,
        )
        self.generate_message = True
```

Reply:
Yes, I tried that, but the base exception `ExceptionGroup` requires a string and a list of exceptions.

https://github.com/home-assistant/core/pull/113844/files#diff-fd4a75ef20756aa31dd323c3ea4c5e279add2e3ee6d8a6cc5eb13681eb7edfa6R85-R88

A custom constructor make all thing complex, so I choose to use the default constructor and the one for `HomeAsssistantError`, that allows `*args: object`.

If you have a better idea, feel free to suggest it.

Reply:
Renaming `translation_key` did the trick